runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -6214,6 +6214,280 @@ static TR::Register *inlineStringHashCode(TR::Node *node, bool isCompressed, TR:
+ * @returns register
+ */
+static TR::Register* inlineIntrinsicIndexOf(TR::Node* node, TR::CodeGenerator* cg, bool isLatin1)
+   {
+   /*
+    *   add dataAddrReg, addressReg, #headerSize                   ; get the starting address of byte array data
+    *   add endReg, dataAddrReg, lengthReg(, lsl #1 if utf16)      ; get the address of the end of the byte array
+    *   subs lengthReg, lengthReg, offsetReg
+    *   b.eq Ldone
+    *   add tmp0Reg, dataAddrReg, #16                              ; save starting address + 16 to tmp0Reg
+    *   add dataAddrReg, dataAddrReg, offsetReg(, lsl #1 if utf16)
+    *
+    *   if isLatin1
+    *     dup vtmp0Reg.16b, charReg
+    *   else
+    *     dup vtmp0Reg.8h, charReg
+    *   subs lengthReg, lengthReg (#8 if isLatin1 else #4)
+    *   ; If the length < 8bytes, then it is not guaranteed that we can read the last 16bytes because the minimum size of the array header is 8bytes.
+    *   b.lt Lessthan8
+    *   subs lengthReg, lengthReg (#8 if isLatin1 else #4)
+    *   b.lt Lresidual
+    * Loop:
+    *   subs lengthReg, lengthReg, (#16 if isLatin1 else #8)
+    *   ldr  vtmp1Reg, [dataAddrReg], #16
+    *   if isLatin1
+    *     cmeq vtmp1Reg.16b, vtmp0Reg.16b, vtmp1Reg.16b
+    *     shrn vtmp1Reg.8b, vtmp1Reg.8h, #4
+    *   else
+    *     cmeq vtmp1Reg.8h, vtmp0Reg.8h, vtmp1Reg.8h
+    *     xtn vtmp1Reg.8b, vtmp1Reg.8h
+    *   umov tmp1Reg, vtmp1Reg.d[0]
+    *   ccmp tmp1Reg, #0, #0, cs                                   ; if lengthReg < 0, clear all condition flags
+    *   b.eq Loop
+    *   cbnz tmp1Reg, Lfound
+    *   cmn lengthReg, (#16 if isLatin1 else #8)
+    *   b.eq Ldone
+    * Lresidual:
+    *   ldurq vtmp1Reg, [endReg, #-16]                             ; Read the last 16bytes. It would not a problem if original length >= 8 because we have a array header space.
+    *   if isLatin1
+    *     cmeq vtmp1Reg.16b, vtmp0Reg.16b, vtmp1Reg.16b
+    *   else
+    *     cmeq vtmp1Reg.8h, vtmp0Reg.8h, vtmp1Reg.8h
+    *   b Lmerge
+    * LessThan8:
+    *   ldurd vtmp1Reg, [endReg, #-8]
+    *   if isLatin1
+    *     cmeq vtmp1Reg.8b, vtmp0Reg.8b, vtmp1Reg.8b
+    *   else
+    *     cmeq vtmp1Reg.4h, vtmp0Reg.4h, vtmp1Reg.4h
+    * Lmerge:
+    *   add dataAddrReg, dataAddrReg, #16
+    *   neg lengthReg, lengthReg, lsl (#2 if isLatin1 else #3)     ; 64 - (4 or 8) * remaining length
+    *   if isLatin1
+    *     shrn vtmp1Reg.8b, vtmp1Reg.8h, #4
+    *   else
+    *     xtn vtmp1Reg.8b, vtmp1Reg.8h, #4
+    *   umov tmp1Reg, vtmp1Reg.d[0]
+    *   lsr tmp1Reg, tmp1Reg, lengthReg                            ; Move the comparison result of remaining data to LSB
+    *   cmp tmp1Reg, #0
+    *   b.eq Ldone
+    * Lfound:
+    *   rbit tmp1Reg, tmp1Reg
+    *   sub dataAddrReg, dataAddrReg, tmp0Reg
+    *   clz tmp1Reg, tmp1Reg
+    *   add resultReg, dataAddrReg, tmp1Reg, lsr #2
+    *   if !isLatin
+    *     lsr resultReg, resultReg, #1
+    * Ldone:
+    *   csinv x0, x0, xzr, ne
```

Comment:
`x0` in this line should be `resultReg`.

Reply:
Fixed it.